spec/models/reservation_spec.rb

```diff
@@ -26,4 +26,38 @@
   it 'validates raise error for invalid status' do
     expect { create(:reservation, status: 'invalid') }.to raise_error(ArgumentError, "'invalid' is not a valid status")
   end
+
+  describe 'associations' do
+    it "ensure to don't create a reservation with an unavailable room" do
+      _available_room = create(:available_double_room)
+      unavailable_room = create(:unavailable_double_room)
+      reservation = create(:reservation, room_id: unavailable_room.id)
+
+      expect(reservation.room.status).to eq('available')
+    end
+
+    it 'validates to assign a guest' do
+      guest = create(:guest)
+      reservation = create(:reservation, guest_id: guest.id)
+
+      expect(reservation.guest).to eq(guest)
+    end
+
+    it 'errors when room is not available' do
+      reservation_one = create(:reservation)
+      reservation_two = build(:reservation, room_id: reservation_one.room_id, checkin_at: reservation_one.checkin_at, checkout_at: reservation_one.checkout_at)
+
+      expect(reservation_two).not_to be_valid
+      expect(reservation_two.errors[:base]).to include('Room is not available for the selected dates')
+    end
+
+    it "errors when there's no available room" do
```

Comment:
```suggestion
    it "validates the availability of alternative rooms" do
```

But as I mentioned before, this should not be a validation.